### PR TITLE
chore(repo): Backport canary package release into v4

### PR DESCRIPTION
**Pull request by anagstef**
## Description

This is a backport PR from #2015 

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/scripts/purge-cache.mjs [hunk 1]
```diff
@@ -12,10 +12,8 @@ const major = parsedVersion.major;
 try {
   if (!prerelease) {
     await fetch(`https://purge.jsdelivr.net/npm/@clerk/clerk-js@${major}/dist/clerk.browser.js`);
-  } else if (prerelease === 'staging') {
-    await fetch(`https://purge.jsdelivr.net/npm/@clerk/clerk-js@staging/dist/clerk.browser.js`);
   } else {
-    await fetch(`https://purge.jsdelivr.net/npm/@clerk/clerk-js@next/dist/clerk.browser.js`);
+    await fetch(`https://purge.jsdelivr.net/npm/@clerk/clerk-js@canary/dist/clerk.browser.js`);
```
Comment (dimkl) [inline packages/clerk-js/scripts/purge-cache.mjs:16]: ❓ Should we use the `clerk-js@${prerelease}/dist/clerk.browser.js` here and in the main (to support the `alpha-v5`  pre-release tag in main).

**Review by dimkl — APPROVED** [review]

File: .github/workflows/release-canary.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ run-name: Canary release from ${{ github.ref_name }}
 on:
   push:
     branches:
-      - main
+      - release/v4
```
Comment (dimkl) [inline .github/workflows/release-canary.yml:7]: 🤔 We probably need to add both branched in main and add a condition for this action to block it in pre-release tags. Let's make this change in another PR.

File: packages/clerk-js/scripts/purge-cache.mjs (hunk 1 above)
Comment (nikosdouvlis) [reply packages/clerk-js/scripts/purge-cache.mjs:16]: We're going to remove the script completely

Comment (anagstef) [reply packages/clerk-js/scripts/purge-cache.mjs:16]: @dimkl this purge script is actually useless and we will be deleting it soon.
